common/ledger/blkstorage/fsblkstorage/blockfile_archiver.go

```diff
@@ -42,9 +43,9 @@ func newBlockfileArchiver(id string, mgr *blockfileMgr) *blockfileArchiver {
 	loggerArchive.Info("newBlockfileArchiver: ", id)
 
 	blockfileDir := filepath.Join(blockarchive.BlockStorePath, ChainsDir, id)
-	arch := &blockfileArchiver{id, mgr, blockfileDir, 1}
+	arch := &blockfileArchiver{id, mgr, blockfileDir, 1, 0}
 
-	if blockarchive.IsArchiver {
+	if blockarchive.IsArchiver || blockarchive.IsClient {
```

Comment:
Do we need "if" condition here?
Is there any other actor other than Archiver and Client?

Reply:
There is as-is role (peer that is neither Archiver nor Client).

Reply:
oh okay.